lib/index.js

```diff
@@ -218,19 +218,30 @@ module.exports = class Breakpoints {
    */
   async #generateAndWriteCSS() {
     const filePath = path.resolve(process.cwd(), this.#config.css.path);
-    let string = "";
-    this.#customProperties.forEach((customProperty) => {
-      string += `--${customProperty.identifier}: "${customProperty.value}";`;
-    });
     const prettierConfig = {
       ...(this.#config.prettier?.configPath
         ? await Breakpoints.getPrettierConfig(this.#config.prettier.configPath)
         : {}),
       parser: "css",
     };
+    let string = "";
+    if (this.#config.css.customMedia) {
+      this.#customProperties.forEach((customProperty) => {
+        // For syntax see postcss-custom-media plugin or W3C draft
+        // Ref: https://www.npmjs.com/package/postcss-custom-media
+        // Ref: https://www.w3.org/TR/mediaqueries-5/#at-ruledef-custom-media
+        string += `@custom-media --${customProperty.identifier} (${customProperty.value});`;
+      });
+    } else {
+      string += `${this.#config.css.element} {`;
+      this.#customProperties.forEach((customProperty) => {
+        string += `--${customProperty.identifier}: "${customProperty.value}";`;
+      });
+      string += `}`;
+    }
```

Comment:
I wonder if custom media queries and custom properties should be exclusionary. Maybe someone needs both?